manim/mobject/coordinate_systems.py

```diff
@@ -1591,10 +1591,30 @@ def get_lines_parallel_to_axis(
         lines1 = VGroup()
         lines2 = VGroup()
         unit_vector_axis_perp_to = axis_perpendicular_to.get_unit_vector()
-        ranges = (
-            np.arange(0, axis_perpendicular_to.x_max, step),
-            np.arange(0, axis_perpendicular_to.x_min, -step),
-        )
+
+        # If they're not on different sides then their product is positive or 0
+        on_same_side = axis_perpendicular_to.x_max * axis_perpendicular_to.x_min > 0
+
+        if on_same_side:
+            if axis_perpendicular_to.x_max >= 0:
+                # Both on positive side
+                positive_end_point = (
+                    axis_perpendicular_to.x_max - axis_perpendicular_to.x_min
+                )
+                ranges = (np.arange(0, positive_end_point, step),)
+            else:
+                # Both on negative side
+                negative_end_point = (
+                    axis_perpendicular_to.x_min - axis_perpendicular_to.x_max
+                )
+                ranges = (np.arange(0, negative_end_point, -step),)
+        else:
+            # On different sides
+            ranges = (
+                np.arange(0, axis_perpendicular_to.x_max, step),
+                np.arange(0, axis_perpendicular_to.x_min, -step),
+            )
```

Comment:
i think I understand what you're trying to do here. But why not, after l. 1597, just removing the second component of `ranges` if x_min is < 0 ? Wouldn't that work ? I don't know if i'm clear.

Reply:
I don't think removing a component (`ranges[0]` or `ranges[1]`) would solve the issue. The reason why is that we also have to change the actual endpoint if `on_same_side` is true.

So if we have the example where `x_min = 2` and `x_max = 5` then our `np.arange` should only contain elements from 2 up until 5. 

Otherwise if we let this code run first: 


```py
        ranges = (
            np.arange(0, axis_perpendicular_to.x_max, step),
            np.arange(0, axis_perpendicular_to.x_min, -step),
        )
```
and and then delete the values going from `x_min` up to zero, then on the right hand side we will now go over 5 units which would be too many (needed to be 3).